extension/src/background/messageListener/freighterApiMessageListener.ts

```diff
@@ -78,10 +80,13 @@ export const freighterApiMessageListener = (
 
   const submitTransaction = async () => {
     const { transactionXdr, network: _network, accountToSign } = request;
+
     const network = _network ?? MAINNET_NETWORK_DETAILS.network;
     const isMainnet = getIsMainnet();
     const { networkUrl } = getNetworkDetails();
-    const transaction = StellarSdk.TransactionBuilder.fromXDR(
+    const isExperimentalModeEnabled = getIsExperimentalModeEnabled();
+    const SDK = isExperimentalModeEnabled ? SorobanSdk : StellarSdk;
+    const transaction = SDK.TransactionBuilder.fromXDR(
       transactionXdr,
       StellarSdk.Networks[network],
```

Comment:
otherwise returns undefined if on futurenet I believe?
```suggestion
      SDK.Networks[network],
```

I tried signing the test XDR using the[ signTransaction playground](https://docs.freighter.app/docs/playground/signTransaction) and I get a "user denied access", could be why? does it sign correctly for you?

Reply:
Ah this is a good catch, but it was actually another place with a similar issue 🤦